### ISSUE TITLE
set_property could not find TARGET pybind11::pybind11. Perhaps it has not yet been created[QUESTION] 

### ISSUE THREAD

**Issue by Akshay-9416** [pybind/pybind11#2642]
I am trying to build real time tools for my project and pybind11 was one essential requirements. I cloned the repo from https://github.com/pybind/pybind11 and executed $catkin_make
and I am getting this below error 

```
CMake Error at pybind11/tools/pybind11Tools.cmake:85 (set_property):
  set_property could not find TARGET pybind11::pybind11.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_tools.cmake:3 (include)
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_catkinConfig.cmake:222 (include)
  /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
  real_time_tools/CMakeLists.txt:14 (find_package)


CMake Error at pybind11/tools/pybind11Tools.cmake:104 (set_property):
  set_property could not find TARGET pybind11::module.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_tools.cmake:3 (include)
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_catkinConfig.cmake:222 (include)
  /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
  real_time_tools/CMakeLists.txt:14 (find_package)


CMake Error at pybind11/tools/pybind11Tools.cmake:112 (set_property):
  set_property could not find TARGET pybind11::pybind11.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_tools.cmake:3 (include)
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_catkinConfig.cmake:222 (include)
  /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
  real_time_tools/CMakeLists.txt:14 (find_package)


CMake Error at pybind11/tools/pybind11Tools.cmake:118 (set_property):
  set_property could not find TARGET pybind11::embed.  Perhaps it has not yet
  been created.
Call Stack (most recent call first):
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_tools.cmake:3 (include)
  /home/branedesk1/Desktop/robot_essentials/devel/share/pybind11_catkin/cmake/pybind11_catkinConfig.cmake:222 (include)
  /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
  real_time_tools/CMakeLists.txt:14 (find_package)


-- Checking for module 'eigen3 >= 3.0.0'
--   Found eigen3 , version 3.3.4
-- Boost version: 1.65.1
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   filesystem
--   system
-- The OS type is rt-preempt
CMake Warning at mpi_cmake_modules/cmake/documentation.cmake:73 (message):
  'BUILD_DOXYGEN_DOCUMENTATION()' is deprecated, please prefer
  'add_doxygen_documentation()'
Call Stack (most recent call first):
  real_time_tools/CMakeLists.txt:132 (build_doxygen_documentation)


-- building doxygen documentation for real_time_tools
-- +++ processing catkin package: 'blmc_drivers'
-- ==> add_subdirectory(blmc_drivers)
-- CMake versions older than 3.12 may warn when looking to Boost components. Custom macros are used to find it.
-- The OS type is rt-preempt
-- Checking for module 'eigen3 >= 3.0.0'
--   Found eigen3 , version 3.3.4
CMake Warning at mpi_cmake_modules/cmake/documentation.cmake:73 (message):
  'BUILD_DOXYGEN_DOCUMENTATION()' is deprecated, please prefer
  'add_doxygen_documentation()'
Call Stack (most recent call first):
  blmc_drivers/CMakeLists.txt:166 (build_doxygen_documentation)


-- building doxygen documentation for blmc_drivers
-- Configuring incomplete, errors occurred!
See also "/home/branedesk1/Desktop/robot_essentials/build/CMakeFiles/CMakeOutput.log".
See also "/home/branedesk1/Desktop/robot_essentials/build/CMakeFiles/CMakeError.log".
Makefile:922: recipe for target 'cmake_check_build_system' failed
make: *** [cmake_check_build_system] Error 1
Invoking "make cmake_check_build_system" failed
```

can someone please point out what the issue is? below is the link to my catkin work space if you want to recreate the error. https://drive.google.com/drive/folders/1NXGYoKpkJfctwIw80R8b40DwJkroZUn7?usp=sharing

EDIT (@YannickJadoul): Surround output with ```

**Comment by henryiii** [comment]
Please follow here: https://github.com/ipab-slmc/pybind11_catkin/pull/13 

Catkin is a specialized build system that is not directly supported by pybind11. I'm happy to give some assistance with it, though, but it's not something any of us use.

**Comment by wxmerkt** [comment]
Hi @Akshay-9416,
Since you're on Melodic, I recommend installing from binaries rather than cloning and building in the workspace: `sudo apt install ros-melodic-pybind11-catkin`.

**Comment by henryiii** [comment]
This has now been fixed in https://github.com/ipab-slmc/pybind11_catkin/pull/13 , so this should be fine. To recap, it wasn't a problem with pybind11, but just changes in pybind11 that catkin's custom tooling hadn't picked up yet, because they needed to support stock Ubuntu 16.04. But 2.6.0 does support stock Ubuntu 16.04 (CMake 3.5 IIRC) when we scaled back the CMake requirements to 3.4. So everything is now up to date!
